app.rb

```diff
@@ -1,8 +1,57 @@
 require 'sinatra/base'
+
 class RockPaperScissors < Sinatra::Base
+
+  get '/' do
+    erb(:index)
+  end
+
   get '/test' do
     'test page'
   end
 
+  post '/game' do
+    @name = params[:name]
+    erb(:game)
+  end
+
+  post '/result' do
+    @user_choice = params[:choice]
+    @opponent_choice = [:rock, :paper, :scissors].sample
+    @result = pick_winner(@user_choice, @opponent_choice)
+    erb(:result)
+  end
+
+  def pick_winner(player, opponent)
```

Comment:
Nested if statement could be optimised better. Maybe use a hash.
Also, put this in a class file and instantiate in the app file to keep it more tidy.